system_tests/pubsub.py

```diff
@@ -169,7 +169,12 @@ def _by_timestamp(message):
         self.assertEqual(message2.data, MESSAGE_2)
         self.assertEqual(message2.attributes['extra'], EXTRA_2)
 
+    def _maybe_emulator_skip(self):
+        if os.getenv(PUBSUB_EMULATOR) is not None:
+            self.skipTest('IAM not supported by Pub/Sub emulator')
+
```

Comment:
Yeah that's sort of what's being protected against, but I realize that isn't a case we need to prepared for. I also initially wanted to use the value of `isinstance(Config.CLIENT.connection.credentials, EmulatorCreds)` but that was more of a mouthful than `os.getenv(PUBSUB_EMULATOR) is not None`.

Maybe I should go back to `isinstance` to emphasize that? Or you prefer the decorator still?